app/src/app/api/v0/assistants/threads/messages/route.ts

```diff
@@ -22,9 +26,28 @@ export const POST = apiHandler(async (req) => {
   });
 
   // Run the thread with streaming output
-  const stream = openai.beta.threads.runs.stream(threadId, {
-    assistant_id: assistantId,
-  });
 
-  return new NextResponse(stream.toReadableStream());
+  try {
+    const stream = openai.beta.threads.runs.stream(
+      threadId,
+      {
+        assistant_id: assistantId,
+      },
+      {
+        signal,
+      },
+    );
+
+    // TODO: prevent a new thread from being added to current run when active
```

Comment:
Do we have to fix this before merging?

Reply:
I need to do a spike on this, we are currently using a version of the sdk that has some of its methods that are depricated. I create a ticket to address that.